test/e2e/apps/job.go

```diff
@@ -817,6 +818,58 @@ var _ = SIGDescribe("Job", func() {
 		gomega.Expect(jobs.Items).To(gomega.BeEmpty(), "Found job %v", jobName)
 	})
 
+	ginkgo.It("should check the ready of a pods", func(ctx context.Context) {
+		backoff := 0
+		ginkgo.By("Creating a job")
+		job := e2ejob.NewTestJob("notTerminate", "all-ready", v1.RestartPolicyNever, parallelism, completions, nil, int32(backoff))
+
+		job, err := e2ejob.CreateJob(ctx, f.ClientSet, f.Namespace.Name, job)
+		framework.ExpectNoError(err, "failed to create job in namespace: %s", f.Namespace.Name)
+
+		ginkgo.By("Ensuring job has ready pods")
+		err = e2ejob.WaitForJobReady(ctx, f.ClientSet, f.Namespace.Name, job.Name, ready)
+		framework.ExpectNoError(err, "failed to ensure job ready in namespace: %s", f.Namespace.Name)
+
+		ginkgo.By("Ensuring pods for job exist")
+		pods, err := e2ejob.GetJobPods(ctx, f.ClientSet, f.Namespace.Name, job.Name)
+		framework.ExpectNoError(err, "failed to get pod list for job in namespace: %s", f.Namespace.Name)
+		podsReady := int32(0)
+		for _, pod := range pods.Items {
+			for _, c := range pod.Status.Conditions {
+				if c.Type == v1.PodReady && c.Status == v1.ConditionTrue {
+					podsReady++
+				}
+			}
+		}
+		gomega.Expect(podsReady).To(gomega.Equal(parallelism), "expected %d running job pods, but got %d", parallelism, podsReady)
+	})
+
+	ginkgo.It("should check the ready of a pods for failed job", func(ctx context.Context) {
+		backoff := 0
+		ginkgo.By("Creating a job")
+		job := e2ejob.NewTestJob("fail", "all-fail", v1.RestartPolicyNever, parallelism, completions, nil, int32(backoff))
+
+		job, err := e2ejob.CreateJob(ctx, f.ClientSet, f.Namespace.Name, job)
+		framework.ExpectNoError(err, "failed to create job in namespace: %s", f.Namespace.Name)
+
+		ginkgo.By("Ensuring job has ready pods")
+		err = e2ejob.WaitForJobReady(ctx, f.ClientSet, f.Namespace.Name, job.Name, 0)
+		framework.ExpectNoError(err, "failed to ensure job ready in namespace: %s", f.Namespace.Name)
+
+		ginkgo.By("Ensuring pods for job exist")
+		pods, err := e2ejob.GetJobPods(ctx, f.ClientSet, f.Namespace.Name, job.Name)
+		framework.ExpectNoError(err, "failed to get pod list for job in namespace: %s", f.Namespace.Name)
+		podsReady := int32(0)
+		for _, pod := range pods.Items {
+			for _, c := range pod.Status.Conditions {
+				if c.Type == v1.PodReady && c.Status == v1.ConditionTrue {
+					podsReady++
+				}
+			}
+		}
+		expectedPodsReady := int32(0)
+		gomega.Expect(podsReady).To(gomega.Equal(expectedPodsReady), "expected %d running job pods, but got %d", expectedPodsReady, podsReady)
```

Comment:
I think the current formulation of the assert is problematic, because the pods are created and are running and ready for a short while. It is possible that we pass the `.WaitForJobReady(ctx, f.ClientSet, f.Namespace.Name, job.Name, 0)` before any pods are Running, but later when we check this condition some pods will be counted as ready. 

Still, as suggested above, I would be in favor of dropping the assert on individual pods.